misc/python/materialize/checks/mzcompose_actions.py

```diff
@@ -60,6 +53,17 @@ def execute(self, e: Executor) -> None:
                 port=6877,
             )
 
+        mz_version = MzVersion(c.sql_query("SELECT mz_version()")[0][0])
+        if self.tag:
+            assert (
+                self.tag == mz_version
+            ), f"Materialize version mismatch, expected {self.tag}, but got {mz_version}"
+        else:
+            version_cargo = MzVersionCargo()
```

Comment:
Yes, I want to make sure we don't pick the wrong version for upgrade tests.

Reply:
The cleaner solution would be to change `Composition.up` to call a `def up_handler()` on the services, but this might be a larger change.

Reply:
Oh yes you are right, the constructor is definitely not the right place for this, I must have suffered a stroke when I suggested that.

Yes you are right such a refactor would be out of scope for this task ... in particular the healthchecks are those awful bashisms and curlisms so it will be unnatural to hack in some SQL there as well.

Please feel free to leave your code as is then.